tests/run/i4404c.scala

```diff
@@ -0,0 +1,23 @@
+import java.lang.reflect.Modifier
+
+trait T {
+  def f = {
+    object O
+  }
+}
+
+class C {
+  def g = {
+    object P
+  }
+}
+
+object Test {
```

Comment:
Can you explain what you are testing here

Reply:
```
trait T {
  def f = {
    object O
  }
}
```
When we compile the above code, the compiler will generate `final lazy <accessor> module def O(): O$` inside method `f`. I tried to check whether the `final` flag of method `O` is dropped correctly during the `LambdaLift` phase through the test.

Reply:
If you want to test that methods lose their `final` modifier, I would replace the object by a normal `def`. I think it is clearer than using an object relying on the fact that it desugars to a lazy val which itself desugars to a method

Reply:
That would be clearer. I'll change it. Thank you!